tests/t541-operator.c

```diff
@@ -0,0 +1,341 @@
+                S[12] * S[3] * S[5];
+
+    S_inv[14] = -S[0]  * S[5] * S[14] +
+                S[0]  * S[6] * S[13] +
+                S[4]  * S[1] * S[14] -
+                S[4]  * S[2] * S[13] -
+                S[12] * S[1] * S[6] +
+                S[12] * S[2] * S[5];
+
+    S_inv[3] = -S[1] * S[6] * S[11] +
+               S[1] * S[7] * S[10] +
+               S[5] * S[2] * S[11] -
+               S[5] * S[3] * S[10] -
+               S[9] * S[2] * S[7] +
+               S[9] * S[3] * S[6];
+
+    S_inv[7] = S[0] * S[6] * S[11] -
+               S[0] * S[7] * S[10] -
+               S[4] * S[2] * S[11] +
+               S[4] * S[3] * S[10] +
+               S[8] * S[2] * S[7] -
+               S[8] * S[3] * S[6];
+
+    S_inv[11] = -S[0] * S[5] * S[11] +
+                S[0] * S[7] * S[9] +
+                S[4] * S[1] * S[11] -
+                S[4] * S[3] * S[9] -
+                S[8] * S[1] * S[7] +
+                S[8] * S[3] * S[5];
+
+    S_inv[15] = S[0] * S[5] * S[10] -
+                S[0] * S[6] * S[9] -
+                S[4] * S[1] * S[10] +
+                S[4] * S[2] * S[9] +
+                S[8] * S[1] * S[6] -
+                S[8] * S[2] * S[5];
+
+    det = 1/(S[0] * S_inv[0] + S[1] * S_inv[4] + S[2] * S_inv[8] + S[3] *
+             S_inv[12]);
+
+    for (CeedInt i = 0; i < 16; i++)
+      S_inv[i] *= det;
+  }
+
+  // Set initial values
+  {
+    CeedScalar nodes[P];
+    CeedLobattoQuadrature(P, nodes, NULL);
+    CeedScalar *u;
+    CeedVectorGetArray(U, CEED_MEM_HOST, &u);
+    for (CeedInt i=0; i<P; i++)
+      for (CeedInt j=0; j<P; j++)
+        u[i*P+j] = nodes[i] + nodes[j];
+    CeedVectorRestoreArray(U, &u);
+  }
+
+  // Apply original operator
+  CeedOperatorApply(op_apply, U, V, CEED_REQUEST_IMMEDIATE);
+
+  // Apply FDM element inverse
+  {
+    // -- Zero corners
+    CeedScalar *v;
+    CeedVectorGetArray(V, CEED_MEM_HOST, &v);
+    v[0]     = 0.0;
+    v[P-1]   = 0.0;
+    v[P*P-P] = 0.0;
+    v[P*P-1] = 0.0;
+    CeedVectorRestoreArray(V, &v);
+
+    // -- Apply FDM inverse to interior
+    CeedOperatorApply(op_inv, V, W, CEED_REQUEST_IMMEDIATE);
+
+    // -- Pick off corners
+    const CeedScalar *w;
+    CeedScalar w_Pi[4];
+    CeedVectorGetArrayRead(W, CEED_MEM_HOST, &w);
+    w_Pi[0] = w[0];
+    w_Pi[1] = w[P-1];
+    w_Pi[2] = w[P*P-P];
+    w_Pi[3] = w[P*P-1];
+    CeedVectorRestoreArrayRead(W, &w);
+
+    // -- Apply inverse of Schur complement
+    CeedScalar v_Pi[4];
+    for (CeedInt i=0; i<4; i++) {
+      CeedScalar sum = 0.0;
+      for (CeedInt j=0; j<4; j++) {
+        sum += w_Pi[j] * S_inv[i*4+j];
+      }
+      v_Pi[i] = sum;
+    }
+
+    // -- Set corners
+    CeedVectorSetValue(V, 0.0);
+    CeedVectorGetArray(V, CEED_MEM_HOST, &v);
+    v[0]     = v_Pi[0];
+    v[P-1]   = v_Pi[1];
+    v[P*P-P] = v_Pi[2];
+    v[P*P-1] = v_Pi[3];
+    CeedVectorRestoreArray(V, &v);
+
+    // -- Apply full FDM inverse again
+    CeedOperatorApply(op_inv, V, U, CEED_REQUEST_IMMEDIATE);
```

Comment:
Ok, this is a single element test for scalar diffusion that replicates the mapping from the FDM element inv to the A_rr inv. It's currently failing, so either there is a problem with the math, a problem with the FDM code, or a problem with the test. That narrows it down :)